src/main.js

```diff
@@ -83,27 +137,102 @@ function addTask() {
      * @todo 1. Send the request to add the task to the backend server.
      * @todo 2. Add the task in the dom.
      */
+    
+    const To_do_task = document.querySelector("#inputTask").value.trim();
+
+    if (To_do_task === '') {
+        displayErrorToast("Enter the field required..");
+        return;
+    }
+
+    displayInfoToast("Please wait...");
+
+    const dataForApiRequest = {
+        title: To_do_task
+    }
+    const headersForApiRequest = {
+        Authorization: 'Token ' + localStorage.getItem('token')
+    }
+
+    axios({
+        headers: headersForApiRequest,
+        url: API_BASE_URL + 'todo/create/',
+        method: 'POST',
+        data: dataForApiRequest,
+    }).then(function ({ data, status }) {
+        displaySuccessToast("Task has been added !!!");
+        document.querySelector("#inputTask").value = '';
+        getTasks();
```

Comment:
You are fetching all the tasks again in case of addition of a task which might make this slow, try to think in the direction of DOM manipulation.